Firestore/core/src/firebase/firestore/api/document_reference.h

```diff
@@ -66,26 +71,29 @@ class DocumentReference {
 
   const std::string& document_id() const;
 
-  FIRCollectionReference* Parent() const;
+  // TODO(varconst) uncomment when core API CollectionReference is implemented.
+  // CollectionReference Parent() const;
 
   std::string Path() const;
 
-  FIRCollectionReference* GetCollectionReference(
-      const std::string& collection_path) const;
+  // TODO(varconst) uncomment when core API CollectionReference is implemented.
+  // CollectionReference GetCollectionReference(
+  //     const std::string& collection_path) const;
 
   void SetData(std::vector<FSTMutation*>&& mutations, Completion completion);
 
   void UpdateData(std::vector<FSTMutation*>&& mutations, Completion completion);
 
   void DeleteDocument(Completion completion);
 
-  void GetDocument(FIRFirestoreSource source, DocumentCompletion completion);
+  void GetDocument(FIRFirestoreSource source,
+                   HandleMaybe<DocumentSnapshot>&& completion);
 
   id<FIRListenerRegistration> AddSnapshotListener(
-      FIRDocumentSnapshotBlock listener, FSTListenOptions* options);
+      HandleMaybe<DocumentSnapshot>&& listener, FSTListenOptions* options);
 
  private:
-  FIRFirestore* firestore_ = nil;
+  Firestore* firestore_ = nullptr;
```

Comment:
I don't have a clear understanding yet how ownership of Firestore will work in pure C++ API. Perhaps this will have to become a shared pointer?